docs/strategies/retry.md

```diff
@@ -127,84 +128,148 @@ The `BackoffType` property's data type is the [`DelayBackoffType`](xref:Polly.De
 
 ### Constant
 
-Even though the `Constant` name could imply that only the `Delay` property is used, in reality all the above listed properties are used.
-
-Step 1: Calculating the base delay:
-
-- If `UseJitter` is set to `false` and `Delay` is specified then `Delay` will be used.
-- If `UseJitter` is set to `true` and `Delay` is specified then a random value is added to the `Delay`.
-  - The random value is between -25% and +25% of `Delay`.
-
-Step 2: Capping the delay if needed:
-
-- If `MaxDelay` is not set then the previously calculated delay will be used.
-- If `MaxDelay` is set and the previously calculated delay is greater than `MaxDelay` then `MaxDelay` will be used.
-
-Step 3: Using the generator if supplied
-
-- If the returned `TimeSpan` of the `DelayGenerator` method call is positive then it will be used.
-- If the returned `TimeSpan` of the `DelayGenerator` method call is negative then it will use the step 2's result.
-- If the `DelayGenerator` method call is `null` then it will use the step 2's result.
-
-> [!NOTE]
-> The `DelayGenerator`'s returned value is not capped with the `MaxDelay`.
+```mermaid
+stateDiagram-v2
+
+    state if_state_step1 <<choice>>
+    state if_state_step2 <<choice>>
+    state if_state_step3 <<choice>>
+    jitter: Should add jitter
+    cap: Should cap delay
+    generator: Should use generator
+    compare: Step1's Delay > MaxDelay
+
+    [*] --> jitter
+    jitter --> cap
+    cap --> generator
+    generator --> [*]
+
+    state jitter {
+        UseJitter --> if_state_step1
+        if_state_step1 --> Delay+Random: true
+        if_state_step1 --> Delay: false
+    }
+    state cap {
+        compare --> if_state_step2
+        if_state_step2 --> MaxDelay: true
+        if_state_step2 --> Step1sDelay: false
```

Comment:
Rather than do this "step 1's thing" approach, how about assign the steps into a value, then just use that value. Then it's more like code.

Something like this (diagram is a bit of a mess, but focus on the connection between the first two boxes and the extra step at the end of box 1):

```mermaid
stateDiagram-v2

    state if_state_step1 <<choice>>
    state if_state_step2 <<choice>>
    state if_state_step3 <<choice>>
    jitter: Should add jitter
    cap: Should cap delay
    compare: X > MaxDelay

    [*] --> jitter
    jitter --> cap
    cap --> *

    state jitter {
        UseJitter --> if_state_step1
        if_state_step1 --> Delay+Random: true
        if_state_step1 --> Delay: false
        Delay+Random --> SetX
        Delay --> SetX
    }
    state cap {
        compare --> if_state_step2
        if_state_step2 --> MaxDelay: true
        if_state_step2 --> Step1sDelay: false
    }
```

Reply:
Something like this:

```mermaid
stateDiagram-v2

    state if_state_step1 <<choice>>
    state if_state_step2 <<choice>>
    state if_state_step3 <<choice>>
    jitter: Should add jitter
    cap: Should cap delay
    generator: Should use generator
    compare: BaseDelay > MaxDelay

    [*] --> jitter
    jitter --> cap
    cap --> generator
    generator --> [*]

    state jitter {
        UseJitter --> if_state_step1
        if_state_step1 --> Delay+Random:true
        if_state_step1 --> Delay: false
        
        Delay+Random --> BaseDelay
        Delay --> BaseDelay
    }
    state cap {
        compare --> if_state_step2
        if_state_step2 --> MaxDelay: true
        if_state_step2 --> Base_Delay: false

        MaxDelay --> NormalizedDelay
        Base_Delay --> NormalizedDelay
    }
    state generator {
        DelayGenerator --> if_state_step3
        if_state_step3 --> GeneratedDelay: positive
        if_state_step3 --> Normalized_Delay: null or negative

        GeneratedDelay --> NextDelay
        Normalized_Delay --> NextDelay
    }
```

Unfortunately the node names are global. So, if I refer to `BaseDelay` from `cap` then it will point to `jitter`'s `BaseDelay`. The workaround was to use different name (with underscore).

Reply:
What about making it a single diagram with decision points, rather than multiple linked boxes?

Reply:
Do you mean something like this?

```mermaid
stateDiagram-v2

    state if_state_step1 <<choice>>
    state if_state_step2 <<choice>>
    state if_state_step3 <<choice>>

    compare: BaseDelay > MaxDelay
    setBase: Set BaseDelay
    setNormalized: Set NormalizedDelay
    setNext: Set NextDelay

    UseJitter --> if_state_step1
    if_state_step1 --> Delay+Random:true
    if_state_step1 --> Delay: false
    
    Delay+Random --> setBase
    Delay --> setBase

    setBase --> compare
    compare --> if_state_step2
    if_state_step2 --> MaxDelay: true
    if_state_step2 --> BaseDelay: false

    MaxDelay --> setNormalized
    BaseDelay --> setNormalized

    setNormalized --> DelayGenerator
    DelayGenerator --> if_state_step3
    if_state_step3 --> GeneratedDelay: positive
    if_state_step3 --> NormalizedDelay: null or negative

    GeneratedDelay --> setNext
    NormalizedDelay --> setNext
    setNext --> [*]
```

Reply:
Yeah, I think that looks clearer as the reader doesn't have to revert back to the earlier part of the diagram.

Reply:
Done